Project-II-Calculator/calculator/src/App.js

```diff
@@ -1,23 +1,62 @@
 import React from 'react';
 import './App.css';
+import CalculatorDisplay from '../src/components/DisplayComponents/CalculatorDisplay';
+import ActionButton from '../src/components/ButtonComponents/ActionButton';
+import NumberButton from '../src/components/ButtonComponents/NumberButton';
+// import ButtonContainer from '../src/components/ButtonComponents/ButtonContainer';
+import * as math from 'mathjs';
 
-const App = () => {
-  return (
-    <div>
-      <h3>Welcome to React Calculator</h3>
-      <p>
-        We have given you a starter project. You'll want to build out your
-        components in their respective files, remove this code and replace it
-        with the proper components.
-      </p>
-      <p>
-        <strong>
-          Don't forget to `default export` your components and import them here
-          inside of this file in order to make them work.
-        </strong>
-      </p>
-    </div>
-  );
-};
+class App extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			input: ''
+		};
+	}
+
+	clickHandler = props => {
+		this.setState({ input: this.state.input + props });
+	};
+	equalHandler = () => {
+		this.setState({ input: math.eval(this.state.input) });
+	};
```

Comment:
This is a nice simple solution to the functionality problem.

Generally .eval shouldn't be used as it allows arbitrary execution of code, this introduces performance issues and security vulnerabilities - in-fact MDN say: "Do not ever use eval!".

However, it's still the most sensible option here I believe - I used it when I did it too :) Just something to be aware of. 

This is regarding the native ```eval``` function, I believe the mathjs library adds some extra security ?!? but it still has a whole section on potential security threats. Just something to be aware of I guess.